syntax = "proto3";

package bloock;

option go_package = "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto";
option java_package = "com.bloock.sdk.bridge.proto";

enum KeyType {
    EcP256k = 0;
    Rsa2048 = 1;
    Rsa3072 = 2;
    Rsa4096 = 3;
}

enum KeyProtectionLevel {
    SOFTWARE = 0;
    HSM = 1;
}

message LocalKey {
    string key = 1;
    KeyType key_type = 2;
    optional string private_key = 3;
}

message ManagedKeyParams {
    KeyProtectionLevel protection = 1;
    KeyType key_type = 2;
    optional string name = 3;
    optional int64 expiration = 4;
}


message ManagedKey {
    string key = 1;
    KeyProtectionLevel protection = 2;
    KeyType key_type = 3;
    optional string name = 4;
    optional int64 expiration = 5;
}